omnigibson/object_states/covered.py

```diff
@@ -94,19 +85,18 @@ def _set_value(self, system, new_value):
         # Currently, we support VisualParticleSystems and PhysicalParticleSystems
         if issubclass(system, VisualParticleSystem):
             # Create the group if it doesn't exist already
-            name = get_element_name_from_system(system)
             if self._visual_particle_group not in system.groups:
                 system.create_attachment_group(obj=self.obj)
 
             # Check current state and only do something if we're changing state
             if self.get_value(system) != new_value:
                 if new_value:
                     # Generate particles
-                    success = system.generate_group_particles_on_object(group=self._visual_particle_group)
-                    # If we succeeded with generating particles (new_value = True), store additional info
-                    if success:
-                        # Store how many particles there are now -- this is the "maximum" number possible
-                        self._n_initial_visual_particles[name] = system.num_group_particles(group=self._visual_particle_group)
+                    success = system.generate_group_particles_on_object(
```

Comment:
shouldn't we first clear all the pre-existing attached particles? i thought i might be added to `generate_group_particles()`, but I didn't see anything there either

Reply:
It's already included in `generate_group_particles_on_object`